Add section to show the active Rust version

matchai opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Implement Rust version section based on the implementation on spaceship.
- Show version section when `Cargo.toml` or `*.rs` are present
- Write tests for the new section
- Fix some existing issues with test teardown

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Closes #68 

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/4658208/46369677-9fc5ef80-c651-11e8-8df6-23288e2a4a82.png)

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->
- [x] I have tested using **MacOS**
- [x] I have tested using **Linux**

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have checked that no other PR duplicates mine
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

matchai committed:
Add teardown steps to tests that required them

matchai committed:
Remove the "_sf" prefix from local variables

matchai committed:
Add the Rust section

matchai committed:
Fix tests that fail after teardown was added

matchai committed:
Add tests for Rust section

matchai committed:
Update documentation for Rust section

matchai commented on functions/__sf_section_golang.fish:
```diff
@@ -33,11 +33,11 @@ function __sf_section_golang -d "Display the current go version if you're inside
 	end
 
-	set -l sf_go_version (go version | awk '{ print substr($3, 3) }')
+	set -l go_version (go version | awk '{ print substr($3, 3) }')
```
Since this variable is local, I removed the `sf_` prefix from it.

Snuggle commented on functions/__sf_section_rust.fish:
```diff
@@ -0,0 +1,40 @@
+		return
+	end
+
+	set -l rust_version (rustc --version | grep --color=never -oE '[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]')
```
This line is compatible with GNU grep (Most Linux systems), but not compatible with ripgrep, toybox's grep or busybox's grep. An if statement check, or using sed might be a good idea.

![image](https://user-images.githubusercontent.com/26250962/46438704-79d64300-c756-11e8-8bf6-a8301af2132c.png)

~~Here is a line that should do exactly the same thing: `rustc --version | sed 's/-.*//' | sed 's/.*\s//'`~~

Here is a line that should do exactly the same thing: 
`rustc --version | sed 's/^[a-z]*\\s//' | sed 's/\\s.*//'`

matchai replied:
Good catch! 👍

Snuggle reviewed (changes requested):
This looks good to me! I'm not satisfied with the grep line, though, since it's not very portable.

matchai committed:
Improve rust version check compatibility

Snuggle committed:
Fixed my broken line of regex

After a little bit of testing and talking with some people, here's a
much much more elegant solution that uses cut. This has been manually
tested on both Linux and in a MacOS virtual machine and should be
compatible with GNU, BSD, Busybox and Toybox coreutils.

matchai commented:
Oh wow! That's a whole lot cleaner.
Thanks! 😄
